rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java

```diff
@@ -422,7 +422,7 @@ private Block createFakePendingBlock(Block best) {
                             emptyUncleHashList, // uncleHash
                             new byte[32], //coinbase
                             new byte[32], // log bloom - from tx receipts
-                            best.getDifficulty(), // difficulty
+                            best.getDifficulty().getBytes(), // difficulty
```

Comment:
Why not use the difficulty object directly? We could use it at block header.

Reply:
Changing the `Block` and `BlockHeader` classes constructor signatures is a huge change. You will notice I didn't do change the `byte[] coinbase` parameter either.
Do you think we can do that as a separate PR? I'm thinking when we tackle the immutable classes would be a good time.

Reply:
It's OK, Please, add a issue to our internal backlog, or this description to any existing item referring immutable block

Reply:
Great! I'll go ahead and test this now.